www/include/configuration/configKnowledge/display-serviceTemplates.php

```diff
@@ -45,32 +45,18 @@
 require_once $centreon_path . '/bootstrap.php';
 $pearDB = $dependencyInjector['configuration_db'];
 
-if (!isset($limit) || !$limit) {
+if (!isset($limit) || !$limit || $limit < 0) {
     $limit = $centreon->optGen["maxViewConfiguration"];
 }
 
-if (isset($_POST['num']) && $_POST['num'] == 0) {
-    $_GET['num'] = 0;
-}
+$orderby = "service_description";
+$order = "ASC";
 
-if (isset($_POST['searchServiceTemplate'])) {
-    if (!isset($_POST['searchHasNoProcedure']) && isset($_GET['searchHasNoProcedure'])) {
-        unset($_REQUEST['searchHasNoProcedure']);
+// Use whitelist as we can't bind ORDER BY values
+if (!empty($_POST['order'])) {
+    if (in_array($_POST['order'], ["ASC", "DESC"])) {
+        $order = $_POST['order'];
     }
-    if (!isset($_POST['searchTemplatesWithNoProcedure']) && isset($_GET['searchTemplatesWithNoProcedure'])) {
-        unset($_REQUEST['searchTemplatesWithNoProcedure']);
-    }
-}
-
-$order = "ASC";
-$orderby = "service_description";
```

Comment:
please restore the orderby => user's filter
please rename using camelCase => orderBy

Reply:
there is only one value available for orderBy => service_description this is why it is hardcoded instead of getting this value in $_POST. But it could be revert if needed.